### PR TITLE
adding option to list all available projects

### PR DESCRIPTION
**Pull request by lhoffjann**
Hi there!

For my rofi integration I needed the ability to simply list all projects. I thought someone else might have a use for this.
It is my first tinkering with rust and there are probably some things missing, so if you have any improvements suggestion, please reach out.

### CODE REVIEW

File: src/commands/list.rs [hunk 1]
```diff
@@ -10,20 +11,30 @@ impl ListCommand {
...
+                    .take(count.unwrap_or(usize::max_value()))
+                    .for_each(|time_entry| println!("{}", time_entry)),
 
+                EntryType::Project => entities
+                    .projects
+                    .values()
+                    .collect::<Vec<&Project>>()
+                    .iter()
+                    .take(count.unwrap_or(usize::max_value()))
+                    .for_each(|projects| println!("{}", projects)),
```
Comment (shantanuraj) [inline src/commands/list.rs:35]: Can simplify this a little bit and avoid the collect call

```suggestion
                    .iter()
                    .take(count.unwrap_or(usize::max_value()))
                    .for_each(|(_, project)| println!("{}", project)),
```

File: src/arguments.rs [hunk 2]
```diff
@@ -74,7 +76,11 @@ pub enum Command {
         cmd: Option<ConfigSubCommand>,
     },
 }
-
+#[derive(Debug, StructOpt)]
+pub enum EntryType {
```
Comment (shantanuraj) [inline src/arguments.rs:80]: I'd rename this to `Entity`, we use `entities` in general to refer to a bunch of the time-entry related entries.
```suggestion
pub enum Entity {
```

File: src/arguments.rs [hunk 3]
```diff
@@ -74,7 +76,11 @@ pub enum Command {
...
+    Project,
+    Entry,
```
Comment (shantanuraj) [inline src/arguments.rs:82]: and this to `TimeEntry` so it's consistent everywhere.
```suggestion
    TimeEntry,
```

**Review by shantanuraj — COMMENTED** [review]
Changes look good, I added some minor comments that'd be nice to have.

I'll resolve the lint issue on the `main` branch

File: src/commands/list.rs (hunk 1 above)
Comment (lhoffjann) [reply src/commands/list.rs]: that makes sense. changed it

File: src/arguments.rs (hunk 2 above)
Comment (lhoffjann) [reply src/arguments.rs]: done!

**Review by shantanuraj — APPROVED** [review]
image

Looks good to me!
Thanks for your contribution

I'll publish a new release asap

**Comment by shantanuraj** [comment]
@lhoffjann This is live on [v0.4.0](https://github.com/watercooler-labs/toggl-cli/releases/tag/v0.4.0) and on [crates.io](https://crates.io/crates/toggl)
Thanks again

**Comment by lhoffjann** [comment]
No Problem. thanks for putting this tool together!

**Pull request merged**